src/core/public/application/application_service.tsx

```diff
@@ -49,48 +54,89 @@ interface StartDeps {
 }
 
 interface AppBox {
-  app: App;
+  app: AppBase;
   mount: AppMounter;
 }
 
+function isLegacyApp(app: AppBox | LegacyApp): app is LegacyApp {
+  return (app as AppBox).mount === undefined;
+}
+
+const allApplicationsFilter = '__ALL__';
+
+interface AppUpdaterWrapper {
+  application: string;
+  updater: AppUpdater;
+}
+
 /**
  * Service that is responsible for registering new applications.
  * @internal
  */
 export class ApplicationService {
-  private readonly apps$ = new BehaviorSubject<ReadonlyMap<string, AppBox>>(new Map());
-  private readonly legacyApps$ = new BehaviorSubject<ReadonlyMap<string, LegacyApp>>(new Map());
+  private started = false;
+  private readonly apps = new Map<string, AppBox | LegacyApp>();
+  private readonly statusUpdaters$ = new BehaviorSubject<Map<symbol, AppUpdaterWrapper>>(new Map());
+
   private readonly capabilities = new CapabilitiesService();
   private mountContext?: IContextContainer<App['mount']>;
 
   public setup({ context }: SetupDeps): InternalApplicationSetup {
     this.mountContext = context.createContextContainer();
 
+    const registerStatusUpdater = (application: string, updater$: Observable<AppUpdater>) => {
+      const updaterId = Symbol();
```

Comment:
seems that we do not use `this.statusUpdaters$` as Map. probably we should use Array/Set instead?

Reply:
As we update the `$statusUpdaters` associated entry when an AppUpdater emits
```typescript
const updaterId = Symbol();
const subscription = updater$.subscribe(updater => {
        const nextValue = new Map(this.statusUpdaters$.getValue());
        nextValue.set(updaterId, {
          application,
          updater,
        });
        this.statusUpdaters$.next(nextValue);
      });
```
I need a way to access the updater$ associated value in `$statusUpdaters`. A set can therefor not be used. An array could, be keeping the insertion index instead of the `updaterId` key in the map, but I though a map made more sense as it was futur-proof against any kind of manipulation we could do on the `statusUpdaters$` value, where keeping an index as id is way more fragile.

Reply:
Also with an array, I need to dissociate the initial `updater$` emission from then others, as the initial should insert a new element in the list, and the next should change the element at position instead

Reply:
I initially idea was to use `application` or `appId` as a identifier to access an updater in an array/set. I see your point and it's perfectly valid 👍